packages/nimble/R/genCpp_sizeProcessing.R

```diff
@@ -566,83 +568,103 @@ sizeRep <- function(code, symTab, typeEnv) {
-    listST <- symTab$getSymbolObject(listDefName, inherits = TRUE)
-    code$type <- "nimbleList"
-    code$sizeExprs <- listST
-    code$toEigenize <- "maybe"
-    code$nDim <- 0
-  }
-  else stop('Error in sizeNewNimbleList: listGenerator not found in parentST', call. = FALSE)
-  
-  asserts <- list()
-  asserts <- c(asserts, recurseSetSizes(code, symTab, typeEnv, useArgs = c(TRUE, rep(FALSE, length(code$args)-1))))
-  if(!(code$caller$name %in% assignmentOperators)){
-    intermediateAsserts <- sizeInsertIntermediate(code$caller, code$callerArgID, symTab, typeEnv)
-    ## intermediateAsserts can potentially have size setting stuff from sizeAssignAfterRecursing.
-    ## Not sure if that would ever happen in this context, but to be safe we'll use last element as the actual intermediate assignment
-    ## Embed the intermediate assignment in a '{' (so insertAssertions will recurse on it) and recurse on it
-    numIntermAsserts <- length(intermediateAsserts)
-    bracketedIntermAssert <- newBracketExpr(intermediateAsserts[numIntermAsserts])
-    exprClasses_setSizes(bracketedIntermAssert, symTab, typeEnv)
-    intermediateAsserts[[numIntermAsserts]] <- bracketedIntermAssert
-    asserts <- c(asserts, intermediateAsserts)
-    return(asserts)
-  }
-  if(length(code$args)>1){   
-    RnewExprs <- list()
-    newExprs <- list()
-    RnfVarExprs <- list()
-    nfVarExprs <- list()
-    exprCounter <- 1
-    originalCode <- code 
-    listElements <- listST$nlProc$symTab$getSymbolNames() ##getSymbolObjects()
-    RlistNameExpr <- nimbleGeneralParseDeparse(originalCode$caller$args[[1]])    
-    for(i in seq_along(listElements)) {
-      thisVarName <- listElements[i]
-      ##      if(!inherits(originalCode$args[[i+1]], 'exprClass') ||  (originalCode$args[[i+1]]$name != "")){  ## skip first arg, which will be name of nlDef, then check if value is ""
-      if(!is.null(originalCode$args[[thisVarName]])) {
-        if(!inherits(originalCode$args[[thisVarName]], 'exprClass') ||  (originalCode$args[[thisVarName]]$name != "")){  ## skip first arg, which will be name of nlDef, then check if value is ""
-          ## nfVar(A, 'x') for whichever element name it's on ('x')
-          ##    RnfVarExprs[[exprCounter]] <- substitute(nfVar(A, X), list(A = RlistNameExpr, X = listElements[[i]]$name))
-          RnfVarExprs[[exprCounter]] <- substitute(nfVar(A, X), list(A = RlistNameExpr, X = thisVarName))
-          ## nfVar(A, 'x') <- y or whatever code was provided (already recursed for size processing)
-          RnewExprs[[exprCounter]] <- substitute(A <- B, list(A = RnfVarExprs[[exprCounter]],
-                                                              B = nimbleGeneralParseDeparse(originalCode$args[[thisVarName]])))
-          exprCounter <- exprCounter + 1
-        }
-      }
+    ## code looks like: nimListDef$new(a = '', b = 12)
+    ## want to change code$caller to :
+    ## { nimList <- nimListDef$new()
+    ## nimList$a <- 10
```

Comment:
It's in a comment.  It's just a hypothetical to illustrate filling in whatever default is defined in the nimbleList.